src/services/session-manager.ts

```diff
@@ -0,0 +1,78 @@
+import { DopplerMvcClient } from './doppler-mvc-client';
+
+type SessionStatus = 'unknown' | 'authenticated' | 'non-authenticated';
+
+interface AppSession {
+  status: SessionStatus;
+}
+
+const noop = () => {};
+
+const defaultSession: AppSession = {
+  status: 'unknown',
+};
+
+export interface SessionManager {
+  session: AppSession;
+}
+
+export class OnlineSessionManager implements SessionManager {
+  private currentSession: AppSession = { ...defaultSession };
+  private handler: (s: AppSession) => void = noop;
+  private dopplerInterval: number | null = null;
+
+  constructor(private dopplerMvcClient: DopplerMvcClient, private keepAliveMilliseconds: number) {}
+
+  public get session() {
+    return this.currentSession;
+  }
+
+  public initialize(handler: (s: AppSession) => void) {
+    this.handler = handler;
+    this.update();
+    this.dopplerInterval = window.setInterval(() => {
+      this.update();
+    }, this.keepAliveMilliseconds);
+  }
+
+  public finalize() {
+    this.handler = noop;
+    if (this.dopplerInterval) {
+      clearInterval(this.dopplerInterval);
+    }
+  }
+
+  private dispatch(data: any) {
+    this.currentSession = { ...defaultSession, ...data };
+    this.handler(this.currentSession);
+  }
+
+  private async update() {
+    try {
+      const dopplerUserData = await this.dopplerMvcClient.getUserData();
+
+      // TODO: do something with dopplerUserData
+
+      // TODO: deal with JWT Token
+      // TODO: get other data related to user
+      this.dispatch({ status: 'authenticated' });
+    } catch (error) {
+      this.logOut();
+    }
+  }
+
+  private logOut() {
+    this.redirect();
+    this.dispatch({ status: 'non-authenticated' });
+  }
+
+  // TODO: move into a dependency
+  private redirect() {
+    const currentUrlEncoded = encodeURI(window.location.href);
+    // TODO: only use redirect on login, not in logout
+    const loginUrl = `${process.env.REACT_APP_API_URL}/SignIn/index?redirect=${currentUrlEncoded}`;
+    window.setTimeout(() => {
+      window.location.href = loginUrl;
```

Comment:
need timeout?

Reply:
I like a timeout to allow update the UI before starting the redirect.

Anyway, I should add a comment to clarify this weird code. 

What is your opinion? keeping the redirect with a clarification comment or starting redirect without give a chance of updating UI?

Reply:
why need updating UI if redirect other side?

Reply:
Supongo que mientras se redirige se mantiene por un ratito la UI, entonces me gustaría que tenga un estado actualizado. ¿Tiene sentido o todos los browsers limpian la pantalla antes de iniciar el proceso de redirección?

Reply:
estoy pensando, pero no se me ocurre, por ahi si es verdad algunos actualizan, igual es un tiempo minimo el que le pones, se llega actualizar?

Reply:
No lo se, me pareció útil por las dudas, ahí agregué un comentario para que quede claro por que estaba eso ahí.